enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -118,53 +98,23 @@ the credentials, I want to stop updating content within the existing pods consum
-reach with this proposal
-- Crafting OLM dependencies between this proposal and [related proposals](https://github.com/openshift/enhancements/pull/214)
-as part of an OLM based install at the time of this writing is immature, at least for some reviewers.
-- However, other reviewers wonder if we are in fact close from having either OLM or CVO operators being able to require
-an OLM operator to be present.
-- Using OLM avoids further increasing the install payload size.
-- Using OLM avoids unexpected hiccups in the feature's installation impacting initial cluster install or cluster upgrade.
-
-#### CSI starting points
-
-**WE ARE GOING WITH EPHEMERAL MODE**
+For the developer preview release, the CSI driver will be installable via OLM.
+This release channel is meant to be a preview only option and allow the CSI driver to be installed on OpenShift outside of the normal release cycle.
 
-**NO SecretProvideClass IS NEEDED**
+When the CSI driver is ready for general availability, it will be installed on OpenShift via the cluster storage operator and an associated delivery operator - see the [CSI driver install proposal](/enhancements/storage/csi-driver-install.md).
+The delivery operator and OLM operator can be one and the same, with appropriate modifications depending on the context.
 
-With those simplifying decisions, a CSI driver with those characteristics still meets our needs.  And the storage
-SMEs recommend that approach to us.
-
-Crafting the CSI driver in this way also helps us deal with SELinux.  Specifically (thanks to @jsafrane for details):
+#### CSI starting points
 
-- when CRI-O starts any container, it updates the SELinux labels on all volumes for that container, where those labels are unique to the container
-- in the context of k8s, multiple containers within a pod have the same filesystem access
-- so, only that specific pod can read files on that volume
-- however, a file can only have one SELinux label
-- so two or more pods cannot access the same file
-- but if each pod has its own file, its own copy, of the shared data, things work easily from a SELinux perspective
-- this is where using ephemeral CSI storage comes in
-- using ephemeral based storage allows for use of a tmpfs (virtual memory filesystem) for storage
-- tmpfs is fast and has negligible overhead
-- it will have to reconstructed if the Pod moves
-- but how it eases dealing with SELinux protection is the bonus we are taking
-- so each pod gets its own tmpfs
-- the kubelet leverages tmpfs in a similar fashion already
+The projected resource CSI driver will provision ephemeral volumes and take advantage of upstream's ephemeral CSI storage mechanisms.
+Ephemeral storage allows the CSI driver to use tmpfs for storage while providing appropriate protection via SELinux labels.
+Each pod will have its own copy of the shared resource that the CSI driver mounts in, and there will be a 1:1 correspondence between the pod and the SELinux label applied to the mounted files.
```

Comment:
"The kubelet manages SELinux labelling for us" was a pertinent phrase I remember from the conversations with @jsafrane, and is a result of the choice made with the first three bullets here

If we are reworking the phrasing here, I say we add that as a way of tying together *why* we are doing it this way .. it also leads into the last bullet below IMO

Reply:
yeah I was trying to capture "work out of the box" with my last comment @jsafrane 

I'm fine with adding that verbiage instead as well @adambkaplan 

But I think some indication that we want to defer to default k8s mechanisms for handling SELinux is a useful thing to do.